### PR TITLE
[Axis Tests] - Rework

### PR DESCRIPTION
**Pull request by bluong**
screen shot 2015-10-16 at 2 25 25 pm

### CODE REVIEW

File: test/axes/axisTests.ts [hunk 1]
```diff
@@ -1,221 +1,237 @@
...
-    baseAxis.orientation("right");
-    assert.isNotNull(baseline.node(), "baseline was drawn");
-    assert.strictEqual(baseline.attr("x1"), "0");
-    assert.strictEqual(baseline.attr("x2"), "0");
-    assert.strictEqual(baseline.attr("y1"), "0");
-    assert.strictEqual(baseline.attr("y2"), String(SVG_HEIGHT));
-
-    svg.remove();
-  });
+    it("updates the layout when updating the margin after rendering", () => {
```
Comment (jtlan) [inline test/axes/axisTests.ts]: Why an extra test for this `describe()` only?

File: test/axes/axisTests.ts [hunk 2]
```diff
@@ -839,15 +871,16 @@ describe("BaseAxis", () => {
             assert.strictEqual(annotationRect.attr("visibility"), "hidden", `rect ${i} outside margin area should be not visible`);
           }
         });
+        axis.destroy();
         svg.remove();
       });
 
       it("shows annotation circles are not hidden regardless if rectangles are hidden", () => {
```
Comment (jtlan) [inline test/axes/axisTests.ts]: This test name isn't quite a sentence.

**Comment by jtlan** [comment]
Some of the `describe()` blocks for orientations don't seem symmetric with each other. Is there a reason for that? Maybe the unique tests could be pulled out into a different block.

**Comment by bluong** [comment]
Employed the usage of wrapping `it` statements into a `forEach` for reusing a single test across orientations

File: test/axes/axisTests.ts [hunk 3]
```diff
@@ -0,0 +1,652 @@
...
+      annotatedTickSet.add(annotatedTick);
+    });
+    assert.strictEqual(axis.content().selectAll(`.${Plottable.Axis.ANNOTATION_LABEL_CLASS}`).size(),
+      annotatedTickSet.size, "only unique annotations rendered");
+
+    axis.destroy();
+    svg.remove();
+  });
+
+  it("renders a duplicate tick in the first tier it can fit in", () => {
```
Comment (jtlan) [inline test/axes/axisTests.ts]: This seems at odds with the previous test. What's the expected behavior?

Comment (bluong) [inline test/axes/axisTests.ts]: The previous test makes sure that only the unique ones render.  However there was an earlier bug where duplicating an annotated tick would offset the tick into a different row.

Ex. Earlier bug had having annotated ticks be [50, 50, 50, 50] render single annotation for 50 but in the 4th row.

Comment (jtlan) [inline test/axes/axisTests.ts]: I think this test should be renamed to make that clear then. "Does not leave extra space for duplicated ticks"?

**Pull request merged**